helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,433 @@
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = createCustomWorkflow(workflowName, "10000", "0");
+    _driver.start(builder.build());
+
+    // Wait until workflow is created.
+    boolean isWorkflowCreated = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1 != null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowCreated);
+
+    _driver.stop(workflowName);
+    // Wait until workflow is stopped.
+    boolean isWorkflowStopped = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1.getWorkflowState() == TaskState.STOPPED);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowStopped);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed created for this
+    // workflow.
+    Assert.assertNotNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNotNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNotNull(_admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+
+    // Stop the Controller
+    _controller.syncStop();
+
+    _driver.delete(workflowName, true);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow.
+    boolean isWorkflowDeleted = TestHelper.verify(() -> {
+      WorkflowConfig wfcfg = _driver.getWorkflowConfig(workflowName);
+      WorkflowContext wfctx = _driver.getWorkflowContext(workflowName);
+      IdealState is = _admin.getResourceIdealState(CLUSTER_NAME, workflowName);
+      return (wfcfg == null && wfctx == null && is == null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowDeleted);
+
+    // Start the Controller
+    String controllerName = CONTROLLER_PREFIX + "_3";
+    _controller = new ClusterControllerManager(ZK_ADDR, CLUSTER_NAME, controllerName);
+    _controller.syncStart();
+
+  }
+
+  @Test(dependsOnMethods = "testDeleteStoppedWorkflowForcefully")
+  public void testDeleteStoppingStuckWorkflowForcefully() throws Exception {
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = createCustomWorkflow(workflowName, "10000", "1000000");
+    _driver.start(builder.build());
+
+    // Wait until workflow is created.
+    boolean isWorkflowCreated = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1 != null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowCreated);
+
+    _driver.stop(workflowName);
+    // Wait until workflow is stopped.
+
+    boolean isWorkflowStopped = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return ((wCtx1.getWorkflowState() == TaskState.STOPPED
+          || wCtx1.getWorkflowState() == TaskState.STOPPING));
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowStopped);
```

Comment:
Per our discussion offline, a "stuck" workflow shouldn't be in the "STOPPED" state, correct?